### ISSUE TITLE
Docs: Navigation Refresh, Search, Accessibility, and Bug Fixes

### ISSUE THREAD

**Issue by lreading** [OWASP/threat-dragon#1544]
**Describe what problem your feature request solves**:  

When reading the documentation, I found myself using the browser's native search function to quickly find information.  Given I'm a maintainer of Threat Dragon, I'm probably _more familiar_ with the docs than our average user.  This inspired me to add a search feature to the Jekyll template.

I've released a new version of the Jekyll template that includes:
- Search
- Accessibility check / fixes (required in CI now)
- Improved header UI (icons with tooltips over long text that doesn't scale well)
- Updated a vulnerable gem
- Fixed a text overflow bug with in left-nav
- Restore mobile nav / UI improvements
- Fixed collapsed menu behavior
- Removed unnecessary files bundled into the template (README, gemspec, etc)

**Describe the solution you'd like**:  

- Update the `owasp-td-jekyll` gem in docs to [v2.1.0](https://github.com/lreading/owasp-td-jekyll/releases/tag/v2.1.0)
- Enable search
- Convert header links to icon/tooltip
- Update light/dark theme switcher tooltip text 

**Declaration**:  

By submitting this issue you have:  

- [x] read the [contribution guide](../blob/main/contributing.md) and agree to the [Code of Conduct](../blob/main/code_of_conduct.md)
- [x] not used agentic or generative AI in creating this feature request

**Additional context**:  

The following screenshots are from the owasp-td-jekyll repo's test harness.  The PR I open will include screenshots highlighting the changes as they relate to TD's docs.

**Search**

<img width="2465" height="1336" alt="Image" src="https://github.com/user-attachments/assets/8ae05007-e977-400e-b2a2-3d8db2f7adc9" />

I am actively avoiding adding new third party dependencies to the theme.  Aside from the recent supply chain issues, the "standard" way of leveraging third party css/javascript in Jekyll is loading them as lib/vendor files.  That pattern makes it difficult to identify known vulns and manage versions.

Jekyll creates a JSON file from the content during build, which is consumed by some light javascript.  The feature set of the search is intentionally limited - I didn't want to pull in a heavy third party dependency for "basic search".  This also means that there is no semantic search, or helpers like a Levenshtein distance correction feature.  This is an intentional tradeoff.

**Header Links**

<img width="1376" height="254" alt="Image" src="https://github.com/user-attachments/assets/18691708-f103-4627-8841-ae4f45173faf" />

The existing header links contain text and an icon.  There was no tooltip.  I added a configurable tooltip config so that we can use _only the icons_. This scales the navbar better on smaller devices, and made room for the search feature.  

**Sidebar Nav Text Overflow**

_Existing version of the theme, fixed in the new version_:

<img width="374" height="577" alt="Image" src="https://github.com/user-attachments/assets/bbc74418-39a3-46dd-9adf-a8581261603e" />

This was a simple CSS/overflow bug that is fixed in the newest version.

**Mobile Nav**

The mobile nav has been broken for a long time due to a missing viewport.  The existing mobile nav was difficult to use and didn't match the theme well.  I restyled it to make it more usable. This screenshot has multiple dropdowns open just for show.

<img width="700" height="1192" alt="Image" src="https://github.com/user-attachments/assets/2a6a4ec9-f314-475e-9cc0-dd26e8849aab" />

**Accessibility**

The repo's actions now run `html-validate`, which provides _some_ coverage for accessibility.  While not perfect, I [[addressed multiple findings](https://github.com/lreading/owasp-td-jekyll/pull/40). This check is required to pass for PRs and releases.  In the future, as I learn more about a11y, I hope to expand the scope.
